test/integration/advertisements_test.rb

```diff
@@ -0,0 +1,202 @@
+require "test_helper"
+require "faker"
+
+class AdvertisementsTest < ActionDispatch::IntegrationTest
+  setup do
+    @campaign = campaigns(:premium)
+    @campaign.update start_date: Date.parse("2019-01-01"), end_date: Date.parse("2019-04-01"), keywords: ENUMS::KEYWORDS.keys.sample(5)
+    @property = properties(:website)
+    @property.update keywords: @campaign.keywords.sample(2)
+    travel_to @campaign.start_date.to_time.advance(days: 15)
+  end
+
+  teardown do
+    travel_back
+  end
+
+  # ----------------------------------------------------------------------------------------------------------
+
+  test "javascript with targeted keywords and undetectable country" do
+    get advertisements_path(@property, format: :js), headers: {"REMOTE_ADDR": "0.0.0.0"}
+    assert response.status == 200
+    assert response.body =~ /CodeFund does not have a advertiser for you at this time/
```

Comment:
We should probably fix the grammar here

"CodeFund does not have *an* advertiser for you at this time"